### PR TITLE
removing --simplify-merges and --show-pulls

### PR DESCRIPTION
**Pull request by zricethezav**
### Description:
Fixes https://github.com/zricethezav/gitleaks/issues/705

### Checklist:

* [x] Does your PR pass tests?
* [-] Have you written new tests for your changes?
* [x] Have you lint your code locally prior to submission?

### CODE REVIEW

**Comment by zricethezav** [comment]
@carlocab heads up I'm removing the dependency on git > 2.27

related discussion https://github.com/Homebrew/homebrew-core/pull/89846#issuecomment-977355147

File: git/git.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func GitLog(source string, logOpts string) (<-chan *gitdiff.File, error) {
 		args = append(args, strings.Split(logOpts, " ")...)
 		cmd = exec.Command("git", args...)
 	} else {
-		cmd = exec.Command("git", "-C", sourceClean, "log", "-p", "-U0", "--full-history", "--simplify-merges", "--show-pulls", "--all")
+		cmd = exec.Command("git", "-C", sourceClean, "log", "-p", "-U0", "--full-history", "--show-pulls", "--all")
```
Comment (carlocab) [inline git/git.go:24]: I'm confused. `--show-pulls` is still here. Am I missing something?

Comment (zricethezav) [reply git/git.go:24]: 🤦🏻 and I committed the wrong thing. Let me fix and release 8.0.2

Comment (zricethezav) [reply git/git.go:24]: @carlocab https://github.com/zricethezav/gitleaks/commit/a37822dffd90ad27fe4aa36c392ea3cc4a6330d8 released with https://github.com/zricethezav/gitleaks/releases/tag/v8.0.2

Thanks for your patience 😄

**Pull request merged**